k9mail/src/main/java/com/fsck/k9/activity/MessageReference.java

```diff
@@ -109,16 +109,19 @@ public String toIdentityString() {
 
     @Override
     public boolean equals(Object o) {
-        if (o instanceof MessageReference == false) {
+        if (!(o instanceof MessageReference)) {
             return false;
         }
         MessageReference other = (MessageReference) o;
-        if ((accountUuid == other.accountUuid || (accountUuid != null && accountUuid.equals(other.accountUuid)))
-                && (folderName == other.folderName || (folderName != null && folderName.equals(other.folderName)))
-                && (uid == other.uid || (uid != null && uid.equals(other.uid)))) {
-            return true;
-        }
-        return false;
+        return equals(other.accountUuid, other.folderName, other.uid, other.flag);
+    }
+
+    @SuppressWarnings("UnusedParameters") // consistency with constructor
```

Comment:
I don't see why we'd need an argument list that matches that of the constructor.